clang/lib/Driver/ToolChains/AVR.cpp

```diff
@@ -651,8 +651,19 @@ void AVR::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   // This is almost always required because otherwise avr-ld
   // will assume 'avr2' and warn about the program being larger
   // than the bare minimum supports.
-  if (Linker.find("avr-ld") != std::string::npos && FamilyName)
```

Comment:
@benshi001 If this line is relocated (moved a bit upwards) so that it is before handling user supplied linker arguments then a user could overwrite the default `-m` option.
As it is the user cannot overwrite the driver generated option.

Reply:
I can remove the new linker options. A user could use `-T` and additional linker options if needed.